vms/platformvm/state/disk_staker_diff_iterator.go

```diff
@@ -0,0 +1,98 @@
+// Copyright (C) 2019-2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package state
+
+import (
+	"encoding/binary"
+	"fmt"
+
+	"github.com/ava-labs/avalanchego/database"
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/utils/hashing"
+)
+
+const (
+	// diffKey = [subnetID] + [inverseHeight] + [nodeID]
+	startDiffKeyLength  = hashing.HashLen + database.Uint64Size
+	diffKeyLength       = startDiffKeyLength + hashing.AddrLen
+	diffKeyHeightOffset = hashing.HashLen
+	diffKeyNodeIDOffset = diffKeyHeightOffset + database.Uint64Size
+
+	// weightValue = [isNegative] + [weight]
+	weightValueLength = 1 + database.Uint64Size
+	weightValueTrue   = 0x01
+)
+
+var (
+	errUnexpectedDiffKeyLength     = fmt.Errorf("expected diff key length %d", diffKeyLength)
+	errUnexpectedWeightValueLength = fmt.Errorf("expected weight value length %d", weightValueLength)
+)
+
+// getStartDiffKey is used to determine the starting key when iterating.
+//
+// Note: the result should be a prefix of [getDiffKey] if called with the same
+// arguments.
+func getStartDiffKey(subnetID ids.ID, height uint64) []byte {
+	key := make([]byte, startDiffKeyLength)
+	copy(key, subnetID[:])
+	packIterableHeight(key[diffKeyHeightOffset:], height)
+	return key
+}
+
+func getDiffKey(subnetID ids.ID, height uint64, nodeID ids.NodeID) []byte {
+	key := make([]byte, diffKeyLength)
+	copy(key, subnetID[:])
+	packIterableHeight(key[diffKeyHeightOffset:], height)
+	copy(key[diffKeyNodeIDOffset:], nodeID[:])
+	return key
+}
+
+func parseDiffKey(key []byte) (ids.ID, uint64, ids.NodeID, error) {
```

Comment:
Technically we never use the `subnetID` returned from this function (except in testing)... But I think it makes sense to keep it... Alternatively we could make this `parseSubnetDiffKey` or something...